app/models/tag.rb

```diff
@@ -72,7 +72,11 @@ def delete_by_digest!(actor)
       return false
     end
 
-    Tag.where(digest: dig).map { |t| t.delete_by!(actor) }
+    tags_destroyed = true
+    Tag.where(digest: dig).find_each do |tag|
+      tags_destroyed &&= tag&.delete_by!(actor)
+    end
+    tags_destroyed
```

Comment:
In `tags_destroyed &&= tag&.delete_by!(actor)` you may have removed tags from a previous iteration, even if this one fails. Thus, you should rename `tags_destroyed` to `success`.